core/renderers/common/constants.js

```diff
@@ -398,3 +441,91 @@ Blockly.blockRendering.ConstantProvider.prototype.shapeFor = function(
       throw Error('Unknown connection type');
   }
 };
+
+/**
+ * Create any DOM elements that this renderer needs (filters, patterns, etc).
+ * @param {!SVGElement} svg The root of the workspace's SVG.
+ * @package
+ */
+Blockly.blockRendering.ConstantProvider.prototype.createDom = function(svg) {
+  /*
+  <defs>
+    ... filters go here ...
+  </defs>
+  */
+  var defs = Blockly.utils.dom.createSvgElement('defs', {}, svg);
```

Comment:
This makes it hard for subclasses to extends this and add additional items to the same ``<defs>`` tag. I don't know if that's important since SVG supports multiple ``<defs>`` tags.

Reply:
Not an issue.